work12/index.js

```diff
@@ -0,0 +1,125 @@
+"use strict";
+const wrap = document.getElementById("js-wrap");
+const ul = document.getElementById("js-list");
+// const button_wrap = document.getElementById("js-button-wrap");
+
+// const url = "https://myjson.dit.upm.es/api/bins/ほげほげajy3";
+// const url = "https://myjson.dit.upm.es/api/bins/86vb";
+// const url = "https://myjson.dit.upm.es/api/bins/eu2f";
+// 下記は、myjson繋がらない時の固定値
+const url = {
+  "data": [
+    {
+      "a": "bookmark",
+      "img": "img/1.png",
+      "alt": "画像１",
+      "text": "ブックマーク"
+    },
+    {
+      "a": "message",
+      "img": "img/2.png",
+      "alt": "画像２",
+      "text": "メッセージ"
+    }
+  ]
+}
+
+async function getData() {
+  try {
+    // const response = await fetch(url);
+    // if (response.ok) {
+    //   const json = await response.json();
+    //   return json;
+    // } else {
+    //   throw new Error(`Server request failed:${response.statusText}`);
+    // }
+    // 下記は、固定値をそのままpromiseの返り値とする
+    const json = url
+
+    return json
+  } catch (e) {
+    console.error(e)
+  }
+}
+
+// async function getData() {
+//   return url
+// }
+
+async function getListData() {
+  let listData;
+  try {
+    listData = await getData();
+  } catch (e) {
+    wrap.textContent = `エラー内容:${e.message}`;
+  } finally {
+    hideLoading();
+  }
+  if (listData.data.length === 0) {
+    wrap.textContent = "data is empty";
+    return;
+  }
+  return listData;
+}
+
```

Comment:
```suggestion
async function getListData() {
  let listData;
  try {
    listData = await getData();
     if (listData.data.length === 0) {
    throw new Error("data is empty");
  }
  return listData;
}
  } catch (e) {
    wrap.textContent = `エラー内容:${e.message}`;
  } finally {
    hideLoading();
  }
```
You can implement these if statement and return(line 58-62) within the `try`.

When `listData.data.length === 0`, call throw here to move to the `catch`, get the contents of throw and display the error statement on the screen based on wrap.textContent = `error content:${e.message}`;.

Reply:
So that's it…
The code is easier to read.

I found my code difficult to read.

commit:[89ba62a](https://github.com/yasu-pro/js-handson/pull/12/commits/89ba62abd758a4c9e4a80de0ab46713a9936f252)

Reply:
I received the URL from "Monaka-san" yesterday, but it seems that there was a discussion that "when the data is empty, you should not throw an error".

Please check the link as it will be posted.

[URL](https://discord.com/channels/726252977928667179/892002061183500348/929020353697710100)

Reply:
I received the URL from "Monaka-san" yesterday, but it seems that there was a discussion that "when the data is empty, you should not throw an error".

Please check the link as it will be posted.

[URL](https://discord.com/channels/726252977928667179/892002061183500348/928656762519846993)

Reply:
Oh, I'm sorry about that. 
I also learned about this part of another student's code which approved, but now I've learned new rule!
I understand that only checking the result of `const listData = await getData();` in this try...catch not but including this if statement. 
Thank you for sharing me :)

Then, you can write here `wrap.textContent = "data is empty" ` as it was. 
Sorry for making you complicated.

Reply:
I learned a lot.

thank you.

commit: [229d1a9](https://github.com/yasu-pro/js-handson/pull/12/commits/229d1a92841530a522f321ea0c0e575c8b398162)